yt/data_objects/construction_data_containers.py

```diff
@@ -815,7 +822,14 @@ def shape(self):
         return tuple(self.ActiveDimensions.tolist())
 
     def _setup_data_source(self):
-        self._data_source = self.ds.region(self.center, self.left_edge, self.right_edge)
+
+        reg = self.ds.region(self.center, self.left_edge, self.right_edge)
+        if self._data_source is None:
+            # always True for YTArbitraryGrid (does not accept a data_source)
```

Comment:
I worry this comment might turn into confusing technical debt if `YTArbitraryGrid` ever receives the missing update.
Maybe add a link to this PR to the comment to say that it holds *as of* this PR, so future readers have an incentive to *check* wether it's still true ?

Reply:
good suggestion, will do.

Reply:
updated!